default/AI/ShipDesignAI.py

```diff
@@ -790,6 +816,102 @@ def update_stats(self, ignore_species=False):
             if self.attacks:
                 self.attacks = foAI.foAIstate.weight_attacks(self.attacks, weapons_grade)
 
+    def _apply_hardcoded_effects(self):
+        """Update stats that can not be read out by the AI yet, i.e. applied by effects.
+
+        This function should contain *all* hardcoded effects for hulls/parts to be considered by the AI
+        to make sure we can easily adjust the values for future balance changes or after implementing a
+        method to read out all stats.
+        """
```

Comment:
So basically once this is live, it's best if any tweaks I make to ship stats I also check this list to make sure I'm not breaking the AI? I can definitely read the code below well enough to edit the numbers, etc.